src/functions/module_array.ts

```diff
@@ -0,0 +1,60 @@
+import { _lt } from "../translation";
+import { AddFunctionDescription, CellValue, MatrixArgValue, PrimitiveArgValue } from "../types";
+import { arg } from "./arguments";
+import { assert, toCellValueMatrix, toNumber } from "./helpers";
+
+// -----------------------------------------------------------------------------
+// EXPAND
+// -----------------------------------------------------------------------------
+export const EXPAND: AddFunctionDescription = {
+  description: _lt("Expands or pads an array to specified row and column dimensions."),
+  args: [
+    arg("array (range<any>)", _lt("The array to expand.")),
+    arg(
+      "rows (number, optional)",
+      _lt("The number of rows in the expanded array. If missing, rows will not be expanded.")
+    ),
+    arg(
+      "columns (boolean, optional)",
+      _lt("The number of columns in the expanded array. If missing, columns will not be expanded.")
+    ),
+    arg("pad_with (any, default=0)", _lt("The value with which to pad.")), // @compatibility: on Excel, pad with #N/A
+  ],
+  returns: ["RANGE<ANY>"],
+  //TODO computeFormat
+  compute: function (
+    array: MatrixArgValue,
+    rows: PrimitiveArgValue | undefined,
+    columns: PrimitiveArgValue | undefined,
+    padWith: PrimitiveArgValue | undefined
+  ): CellValue[][] {
+    const _rows = rows !== undefined ? toNumber(rows) : array[0].length;
+    const _columns = columns !== undefined ? toNumber(columns) : array.length;
+    const _padWith = padWith !== undefined && padWith !== null ? padWith : 0; // TODO : Replace with #N/A errors once it's supported
+
+    assert(
+      () => _rows >= array[0].length,
+      _lt(
+        "The rows arguments (%s) must be greater or equal than the number of rows of the array.",
+        _rows.toString()
+      )
+    );
+    assert(
+      () => _columns >= array.length,
+      _lt(
+        "The columns arguments (%s) must be greater or equal than the number of columns of the array.",
+        _columns.toString()
+      )
+    );
+
+    const result: MatrixArgValue = [];
+    for (let col = 0; col < _columns; col++) {
+      result[col] = [];
+      for (let row = 0; row < _rows; row++) {
+        result[col][row] = array[col]?.[row] || _padWith;
+      }
+    }
+    return toCellValueMatrix(result);
```

Comment:
I think that's overkill.

If we reconstruct an array, in the previous steps by reassigning all the values, then we can handle the undefined case there.

Here we go through the matrix a second time for nothing

Reply:
(note that here ```àrray``` contains only values having the type CellValue or undefined, not null)